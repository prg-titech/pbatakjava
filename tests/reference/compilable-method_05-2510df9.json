{
    "basename": "compilable-method_05-2510df9",
    "cmd": "{infile}",
    "infile": "tests/method_05/build.sh",
    "infile_hash": "6fc558c381e9a7bb260ed9e39e4f0a7b2b4b893bef0755f2f9d62355",
    "stdout": "compilable-method_05-2510df9.stdout",
    "stdout_hash": "3db55cd7afe7fdc7ca973aa64b45d86cb63caae74a7bb667edf3a4a8",
    "stderr": null,
    "stderr_hash": null,
    "returncode": 0
}