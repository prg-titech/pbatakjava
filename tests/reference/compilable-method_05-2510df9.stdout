
public interface Point {
  public int get__x();

  public int set__x(int toAssign);

  public int get__y();

  public int set__y(int toAssign);

  public int getX();

  public boolean testBoolean();

  public char testChar();

  public float get__x__float();

  public float set__x(float toAssign);

  public float get__y__float();

  public float set__y(float toAssign);

  public float getY();
}

public interface Point_Factory {
  public Point make(int x, int y);
}

public class Point_v1 implements Point {
  public int x;

  public int y;

  public Point_v1(int x, int y) {
    this.x = x;
    this.y = y;
  }

  public int getX() {
    return this.x;
  }

  public boolean testBoolean() {
    return true;
  }

  public char testChar() {
    return 'k';
  }

  public int get__x() {
    return this.x;
  }

  public int set__x(int toAssign) {
    return this.x = toAssign;
  }

  public int get__y() {
    return this.y;
  }

  public int set__y(int toAssign) {
    return this.y = toAssign;
  }

  public float get__x__float() {
    return 0;
  }

  public float set__x(float toAssign) {
    return 0;
  }

  public float get__y__float() {
    return 0;
  }

  public float set__y(float toAssign) {
    return 0;
  }

  public float getY() {
    return 0;
  }
}

public class Point_v1_Factory implements Point_Factory {
  public Point make(int x, int y) {
    return new Point_v1(x, y);
  }
}

public class Point_v2 implements Point {
  public float x;

  public float y;

  public Point_v2(int x, int y) {
    this.x = x;
    this.y = y;
  }

  public float getY() {
    return this.y;
  }

  public float get__x__float() {
    return this.x;
  }

  public float set__x(float toAssign) {
    return this.x = toAssign;
  }

  public float get__y__float() {
    return this.y;
  }

  public float set__y(float toAssign) {
    return this.y = toAssign;
  }

  public int get__x() {
    return 0;
  }

  public int set__x(int toAssign) {
    return 0;
  }

  public int get__y() {
    return 0;
  }

  public int set__y(int toAssign) {
    return 0;
  }

  public int getX() {
    return 0;
  }

  public boolean testBoolean() {
    return false;
  }

  public char testChar() {
    return 0;
  }
}

public class Point_v2_Factory implements Point_Factory {
  public Point make(int x, int y) {
    return new Point_v2(x, y);
  }
}
